### PR TITLE
Param adding additions

### PR DESCRIPTION
**Pull request by milljm**
Works. Now we need a unittest to see how many things this might be breaking...

Refs #26215

### CODE REVIEW

File: python/TestHarness/JobDAG.py [hunk 1]
```diff
@@ -237,6 +241,32 @@ def _doRaceConditions(self):
...
+                and tester.specs[param] != prereq_tester.specs[param]
+                and not prereq_tester.checkRunnableBase(self.options)):
+                # ...add param value to prereq tester in an attempt to allow prereq tester to launch
+                prereq_tester.specs[param] = tester.specs[param]
+
+                # After running checkRunnableBase it is necessary to 'reset' some tester attributes
+                # TODO: Maybe add a feature to checkRunnableBase, indicating not to cache settings
+                #       so all we have to do is add the missing param
+                prereq_tester.setStatus(prereq_tester.no_status)
+                prereq_tester.addCaveats(f'{param} param added to satisfy dependency')
```
Comment (loganharbour) [inline python/TestHarness/JobDAG.py:267]: Is also making it a heavy test the only way to get it to run?

**Review by loganharbour — COMMENTED** [review]
Just a logic question

File: python/TestHarness/JobDAG.py (hunk 1 above)
Comment (milljm) [reply python/TestHarness/JobDAG.py:267]: Yeah I like that. What I had was a bit long winded

**Comment by milljm** [comment]
Closing in favor of #26329

**Pull request closed without merging**